### DISCUSSION TITLE
Question about #nullable enable

### DISCUSSION THREAD

**Discussion by tranb3r** [CommunityToolkit/dotnet#656] [Q&A] [answered]
Hi,

The generated code for `ObservableProperty` has the `#nullable enable` directive.
Does it mean that the code where the property field is defined also has to have the same directive in order to work?

Example:
```
[ObservableProperty]
private string _name;

private void Test()
{
    if (Name == null)
    // ....
}
```

This code generates a warning in visual studio: `Expression is always false`.
But if `_name` is not initialized, `Name` is `null` when testing it, and at runtime the condition returns `True`.

So, can I safely assume that the generated code will work, even if visual studio shows warnings ?

**Comment by Sergio0694** [comment, marked as answer]
That is correct, you're getting the warning because you've marked the property as not nullable. You can either ignore the warning (since the property can in fact be null), or enable nullability annotations in that file (or in your whole project, if possible).
